### PR TITLE
Handle mentions

### PR DESCRIPTION
**Pull request by karx1**
# Pull Request

## Description:

This PR lets the bot fire if any of the loved people are mentioned. 
This means, if you mention "@Matt", then the bot would respond, "Matt is Amazing!"

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -29,28 +29,42 @@ const token = process.env.TOKEN;
 let timer = 0;
 
 // condition validation
+const checkMentions = (message: Message, checkId: string): boolean => {
+  let match = false;
+  message.mentions.mentions.users.map(user => {
```
Comment (nhcarrigan) [inline src/index.ts]: Double `mentions` property seems incorrect.

File: src/index.ts [hunk 2]
```diff
@@ -29,28 +29,42 @@ const token = process.env.TOKEN;
...
+    if (user.id === checkId) {
+      match = true;
+      break;
+    }
+  });
+  return match;
+};
```
Comment (nhcarrigan) [inline src/index.ts]: I think, rather than creating a map, we could use `.find` or `.some` (find would be more performant, likely). Then we don't need to break early or even assign a boolean variable.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (karx1) [reply src/index.ts]: Whoops, sorry, fixing that

File: src/index.ts (hunk 2 above)
Comment (karx1) [reply src/index.ts]: `find` returns the actual value, not a boolean. I suppose I could check for null?

Comment (nhcarrigan) [reply src/index.ts]: It either returns the first value that passes the callback function, or `undefined` if no value matches.

So we could shorten this to:
```ts
return !!message.mentions.find()
```

Comment (karx1) [reply src/index.ts]: Alright, will do

**Comment by karx1** [comment]
@nhcarrigan Instead of `find`, I used the built-in method `has` of `message.mentions`

File: src/index.ts [hunk 3]
```diff
@@ -29,28 +29,35 @@ const token = process.env.TOKEN;
 let timer = 0;
 
 // condition validation
+const checkMentions = (message: Message, checkId: string): boolean => {
+  return message.mentions.has(checkId);
+};
+
```
Comment (nhcarrigan) [inline src/index.ts:35]:
```suggestion
const checkMentions = (message: Message, checkId: string): boolean => {
  return message.content.toString().includes(checkId);
};
```
Pulling this down locally, the `has` method isn't firing correctly on the `id` string. It looks like the docs say it requires a UserResolvable object. We could optionally fetch the users based on ID if needed. However, instead of that, if we cast the message content into a string then mentions are converted to `<@!id>` format and the `includes` method will catch them correctly.

Thoughts?

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (karx1) [reply src/index.ts:35]: @nhcarrigan Looking at the UserResolvable docs,
image
We can pass in a user object or a `Snowflake`, which is an id. Not sure why it's not working though.

Has the casting method worked for you in your testing?

Comment (nhcarrigan) [reply src/index.ts:35]: The casting method did work for me, yes.

The problem with UserResolvable is that a Snowflake is different from a string - we'd have to parse the `id` string into a Snowflake object.

Comment (karx1) [reply src/index.ts:35]: Oh, ok. I'll commit your suggestion then.

**Review by nhcarrigan — APPROVED** [review]
LGTM~! 🥳 

Thank you for your pull request! We look forward to additional contributions!
